src/lesson-2-1-demo/ClassComponents.jsx

```diff
@@ -1,96 +1,62 @@
 import { faker } from "@faker-js/faker";
-import { createContext, Component } from "react";
+import {
+  createContext,
+  Component,
+  useContext,
+  useState,
+  useEffect,
+} from "react";
 
-export class HelloWorld extends Component {
-  render() {
-    return <p>Hello, World!</p>;
-  }
-}
-
-export class Hello extends Component {
-  render() {
-    const { name } = this.props;
-    return <p>Hello, {name}!</p>;
-  }
-}
+export const HelloWorld = () => <p>Hello, World!</p>;
 
-export class Counter extends Component {
-  // Aside: Notice how the constructor accepts props as the first argument
-  constructor(props) {
-    // Notice how we MUST call super(props)
-    super(props);
-    this.state = { count: 0 };
-  }
-
-  incrementCount() {
-    this.setState({ count: this.state.count + 1 });
-  }
+export const Hello = ({ name }) => <p>Hello, {name}!</p>;
 
-  render() {
-    return (
-      <div>
-        <p>Count: {this.state.count}</p>
-        <button id="lmao" onClick={() => this.incrementCount()}>
-          Increment
-        </button>
-      </div>
-    );
-  }
-}
-
-export class Timer extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { interval: null, start: performance.now(), elapsed: 0 };
-  }
+export const Counter = () => {
+  const [count, setCount] = useState(0);
+  const incrementCount = () => setCount(count + 1);
+  return (
+    <div>
+      <p>Count: {count}</p>
+      <button onClick={incrementCount}>Increment</button>
+    </div>
+  );
+};
 
-  componentDidMount() {
+export const Timer = ({ fps }) => {
+  const [timerInterval, setTimerInterval] = useState(null);
+  const [start] = useState(performance.now());
+  const [elapsed, setElapsed] = useState(0);
+  useEffect(() => {
```

Comment:
Lifecycle methods are converted to hooks using this conversion guide

Class Component Lifecycle | useEffect Hook
-- | --
componentDidMount | useEffect(() => { /* logic here */ }, [])
componentDidUpdate | useEffect(() => { /* logic here */ })
componentWillUnmount | useEffect(() => { return () => { /* logic here */ } })